tensorflow/python/eager/function.cc

```diff
@@ -0,0 +1,84 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <Python.h>
+
+#include "pybind11/pybind11.h"
+#include "pybind11/stl_bind.h"
+
+struct ConcreteFunction;  // Forward declaration.
+
+// TODO(jlchu): Migrate Python characteristics to C++
+
+namespace tensorflow {
+
+namespace py = pybind11;
+
+py::module* np_arrays = new py::module(
+    py::module::import("tensorflow.python.ops.numpy_ops.np_arrays"));
+py::module* nest = new py::module(
+    py::module::import("tensorflow.python.util.nest"));
+
+struct ConcreteFunction {
+  ConcreteFunction() {};
+  py::object _build_call_outputs(py::object result,
+                                 py::object structured_outputs,
+                                 bool _ndarrays_list,
+                                 bool _ndarray_singleton);
+};
+
+py::object ConcreteFunction::_build_call_outputs(py::object result,
```

Comment:
Can we have a unit test for this?

Reply:
For now, this function is simply called by the python version in `function.py` - would existing tests for the python then cover this, or should there be something more? Will discuss with @sun51 .

Reply:
Update: C++ is currently added as standalone, will not be called from Python until execution is faster